### ISSUE TITLE
How to run locally directly from the repo?

### ISSUE THREAD

**Issue by trusktr** [onivim/oni#1626]
It's not very clear.

I tried initing sub modules, then `yarn install && npm run build && npm start` which opens the window, but the windows tells me to wait for Webpack and that I may need to restart it.

I keep trying to restart it but that message never goes away.

What's the official steps to run the editor from the git repo?

**Comment by trusktr** [comment]
I'm on macOS by the way.

**Comment by trusktr** [comment]
For now, I am able to use the `.app` for macOS, but still want to know how to start manually.

**Comment by CrossR** [comment]
They would be the steps I'd use to get it running, just using `npm run start` instead which I think does the same thing.

It'll sit on the waiting for webpack for a second or so, but then moves on and refreshes.

I know some users had issues where we weren't refreshing after the compilation had finished, but I thought it had been sorted.

There is an issue here about it : #1345.
It might be worth trying the fix in there, and if that works, we can look at sorting that issue. (Though now that I'm reading that ticket again I realise @someguynamedmatt is having the issue still)

**Comment by someguynamedmatt** [comment]
@CrossR yep, I’m still seeing that too. I tried to work with a fix which included `window.location.relod()` when the webpack bundle was loaded, by that didn’t totally work.

**Comment by someguynamedmatt** [comment]
@trusktr Yeah, refresh the electron instance (like you would with”normal” Chrome) on Mac cmd-r

**Comment by trusktr** [comment]
`npm start` by itself threw an error. Then I tried `npm run build && npm start` which got me further (to the state that tells me to refresh above, though refreshing leads me back to state that tells me to refresh, and refreshing never works).

**Comment by trusktr** [comment]
On commit d7b1c4fd (which passes all CI checks), I tried `yarn install && yarn start` which results in the following output with two errors:

```
[1] Project is running at http://localhost:8191/
[1] webpack output is served from http://localhost:8191/
[0] 2018-02-26 20:07:23.555 Electron Helper[72054:2682633] Couldn't set selectedTextBackgroundColor from default ()
[1] ts-loader: Using typescript@2.7.1 and /Users/joe.pea/src/oni/browser/tsconfig.json
[2] 8:07:25 PM - Compilation complete. Watching for file changes.
[2]
[2]
[1] Hash: a9eb5d6420497f2c5421
[1] Version: webpack 3.5.3
[1] Time: 18103ms
[1]        Asset     Size                                                                          Chunks                    Chunk Names
[1] 21.bundle.js  1.91 MB                                                                          21, 22  [emitted]  [big]
[1]  0.bundle.js  12.7 kB                                                                               0  [emitted]
[1]  2.bundle.js  7.07 MB  2, 0, 3, 5, 6, 7, 8, 9, 13, 15, 17, 18, 22, 24, 25, 26, 27, 28, 30, 31, 32, 37  [emitted]  [big]
[1]  3.bundle.js  6.08 MB                                                     3, 0, 7, 15, 18, 22, 25, 32  [emitted]  [big]
[1]  4.bundle.js  3.66 MB                                                        4, 0, 15, 22, 25, 27, 32  [emitted]  [big]
[1]  5.bundle.js  4.71 MB                                                     5, 0, 7, 15, 18, 22, 25, 32  [emitted]  [big]
[1]  6.bundle.js   4.7 MB                                                     6, 0, 7, 15, 18, 22, 25, 32  [emitted]  [big]
[1]  7.bundle.js   4.7 MB                                                        7, 0, 15, 18, 22, 25, 32  [emitted]  [big]
[1]  8.bundle.js  2.95 MB                                                            8, 0, 15, 22, 25, 32  [emitted]  [big]
[1]  9.bundle.js  2.75 MB                                                                   9, 22, 26, 31  [emitted]  [big]
[1] 10.bundle.js   2.9 MB                                                           10, 0, 15, 22, 25, 32  [emitted]  [big]
[1] 11.bundle.js  2.87 MB                                                           11, 0, 15, 22, 25, 32  [emitted]  [big]
[1] 12.bundle.js  2.81 MB                                                               12, 0, 15, 22, 25  [emitted]  [big]
[1] 13.bundle.js   2.8 MB                                                                       13, 0, 32  [emitted]  [big]
[1] 14.bundle.js   800 kB                                                                              14  [emitted]  [big]
[1] 15.bundle.js  2.26 MB                                                                       15, 0, 22  [emitted]  [big]
[1] 16.bundle.js   577 kB                                                                           16, 0  [emitted]  [big]
[1] 17.bundle.js   559 kB                                                                              17  [emitted]  [big]
[1] 18.bundle.js  2.08 MB                                                                          18, 22  [emitted]  [big]
[1] 19.bundle.js   634 kB                                                                       19, 0, 25  [emitted]  [big]
[1] 20.bundle.js  1.92 MB                                                                          20, 22  [emitted]  [big]
[1]  1.bundle.js  8.14 MB                       1, 0, 3, 7, 9, 13, 15, 18, 22, 25, 26, 27, 30, 31, 32, 37  [emitted]  [big]
[1] 22.bundle.js  1.91 MB                                                                              22  [emitted]  [big]
[1] 23.bundle.js   576 kB                                                                              23  [emitted]  [big]
[1] 24.bundle.js   538 kB                                                                          24, 31  [emitted]  [big]
[1] 25.bundle.js   431 kB                                                                           25, 0  [emitted]  [big]
[1] 26.bundle.js   401 kB                                                                              26  [emitted]  [big]
[1] 27.bundle.js   400 kB                                                                              27  [emitted]  [big]
[1] 28.bundle.js   385 kB                                                                              28  [emitted]  [big]
[1] 29.bundle.js   381 kB                                                                              29  [emitted]  [big]
[1] 30.bundle.js   379 kB                                                                              30  [emitted]  [big]
[1] 31.bundle.js   386 kB                                                                              31  [emitted]  [big]
[1] 32.bundle.js   358 kB                                                                           32, 0  [emitted]  [big]
[1] 33.bundle.js   107 kB                                                                              33  [emitted]
[1] 34.bundle.js   140 kB                                                                              34  [emitted]
[1] 35.bundle.js    41 kB                                                                           35, 0  [emitted]
[1] 36.bundle.js  18.1 kB                                                                           36, 0  [emitted]
[1] 37.bundle.js  27.3 kB                                                                              37  [emitted]
[1] 38.bundle.js  21.5 kB                                                                              38  [emitted]
[1] 39.bundle.js   189 kB                                                                              39  [emitted]
[1] 40.bundle.js  77.3 kB                                                                              40  [emitted]
[1]    bundle.js  1.66 MB                                                                              41  [emitted]  [big]  main
[1]   [98] ./browser/src/Log.ts 1.02 kB {41} [built]
[1]   [99] ./browser/src/Services/CommandManager.ts 2.17 kB {0} {1} {2} {3} {4} {5} {6} {7} {8} {10} {11} {12} {13} {15} {16} {19} {25} {32} {35} {36} [built]
[1]  [103] ./browser/src/Utility.ts 4.94 kB {41} [built]
[1]  [107] ./browser/src/Performance.ts 524 bytes {41} [built]
[1]  [119] ./node_modules/strip-ansi/index.js 161 bytes {41} [built]
[1]  [133] multi (webpack)-dev-server/client?http://localhost:8191 ./browser/src/index.tsx 40 bytes {41} [built]
[1]  [134] (webpack)-dev-server/client?http://localhost:8191 5.83 kB {41} [built]
[1]  [200] ./browser/src/index.tsx 11.3 kB {41} [built]
[1]  [300] ./browser/src/Services/Configuration/ConfigurationCommands.ts 577 bytes {21} [built]
[1]  [309] ./browser/src/Services/Explorer/index.tsx 236 bytes {4} [built]
[1]  [310] ./browser/src/Services/Search/index.tsx 3.42 kB {10} [built]
[1]  [311] ./browser/src/Services/Browser/index.tsx 7.01 kB {23} [built]
[1]  [312] ./browser/src/Services/AutoClosingPairs.ts 7.36 kB {38} [built]
[1]  [313] ./browser/src/Services/Bookmarks/index.ts 1.41 kB {11} [built]
[1]  [314] ./browser/src/Plugins/PluginSidebarPane.tsx 3.45 kB {12} [built]
[1]     + 1096 hidden modules
[1]
[1] WARNING in ./node_modules/cross-spawn/index.js
[1] Module not found: Error: Can't resolve 'spawn-sync' in '/Users/joe.pea/src/oni/node_modules/cross-spawn'
[1]  @ ./node_modules/cross-spawn/index.js 32:26-47
[1]  @ ./node_modules/shell-env/node_modules/execa/index.js
[1]  @ ./node_modules/shell-env/index.js
[1]  @ ./browser/src/Plugins/Api/Process.ts
[1]  @ ./browser/src/Services/Language/LanguageClientProcess.ts
[1]  @ ./browser/src/Services/Language/index.ts
[1]  @ ./browser/src/index.tsx
[1]  @ multi (webpack)-dev-server/client?http://localhost:8191 ./browser/src/index.tsx
[1] webpack: Compiled with warnings.

**Comment by someguynamedmatt** [comment]
I see those as well in my terminal output. That shouldn't be stopping you. Here's what you can try (this is frustrating, I know. Bare with me):

1) make sure you're using node 9 (the docs say 8, but 9 hasn't given me trouble)

2) `rm -r node_moduls && npm i && npm link && yarn install && npm run build && npm run start`

     - That's a lot, I know. I like to do everything at once so I can go get a coffee or whatever

That _should_ get you going

**Comment by trusktr** [comment]
That should be completely unnecessary. Looks like the `scripts` need some love.

**Comment by someguynamedmatt** [comment]
It is and they do. Feel free to help out!

**Comment by trusktr** [comment]
Okay, after running that long command, same thing about telling me to refresh. This time, I tried refreshing several times, and then finally after doing it repeatedly did it load into the editor.

**Comment by Akin909** [comment]
@trusktr there's an issue somewhere @bryphe raised re having to reload the editor manually in dev mode to get it to boot up not sure anyone's had a look yet

**Comment by bryphe** [comment]
#1663 should smooth this out. As an alternative, if the `webpack-dev-server` is causing problems, you can try:
- `npm run build-debug`
- `npm run launch`

**Comment by bryphe** [comment]
This should be solved now with #1345, so I'll close this out. Feel free to log a new issue if there are still problems with the dev experience.
